### PR TITLE
Handle sdk load errors

### PR DESCRIPTION
**Pull request by ihorserba**
Handle SDK loading errors.
Fixed the session connect promise to be rejected if failed to connect.

### CODE REVIEW

File: sdks/js/src/classes/sdk.js [hunk 1]
```diff
@@ -37,8 +37,10 @@ class Sdk {
    *                           or provide class function to be used as a custom player, decoder, recorder or encoder
    *                           (see <a href="">examples</a>).
    *
-   * @param {function} [userCallback] User callback to fire when sdk parts required by this init call are loaded
-   * @return {promise} Promise that resolves when sdk parts required by this init call are loaded
+   * @param {function} [userCallback] User callback to fire when sdk parts required by this init call are loaded,
+   *                                  or any of them was failed to load.
```
Comment (vocoded) [inline sdks/js/src/classes/sdk.js:41]:
```suggestion
   *                                  or any of them failed to load.
```

File: sdks/js/src/classes/sdk.js [hunk 2]
```diff
@@ -37,8 +37,10 @@ class Sdk {
...
+   * @return {promise} Promise that resolves when sdk parts required by this init call are loaded,
+   *                   or rejects if any of the required components was failed to load.
```
Comment (vocoded) [inline sdks/js/src/classes/sdk.js:43]:
```suggestion
   *                   or rejects if any of the required components failed to load.
```

File: sdks/js/src/classes/sdk.js [hunk 3]
```diff
@@ -96,23 +106,39 @@ ZCC.Sdk.init({
...
 
     $script(scriptsToLoad, 'bundle');
     $script.ready('bundle', () => {
+      const library = Utils.getLoadedLibrary();
+      for (const name of scriptsNames) {
+        if (library && library[name]) {
+          continue;
+        }
+        if (typeof userCallback === 'function') {
+          userCallback.apply(userCallback, 'Unable to load ' + name);
```
Comment (vocoded) [inline sdks/js/src/classes/sdk.js:137]: apply takes a second argument that's an array. use `call` if you want to send the arguments one by one

File: sdks/js/src/classes/session.js [hunk 4]
```diff
@@ -154,6 +154,7 @@ session.connect(function(err, result) {
          * @event Session#session_disconnect
          * @param {string} error Error description
          */
+        dfd.reject(err);
```
Comment (vocoded) [inline sdks/js/src/classes/session.js:157]: does this need to be before the emit?

**Review by vocoded — CHANGES_REQUESTED** [review]
couple small things

File: sdks/js/src/classes/sdk.js (hunk 1 above)
Comment (ihorserba) [reply sdks/js/src/classes/sdk.js]: Changed

File: sdks/js/src/classes/session.js [hunk 5]
```diff
@@ -172,8 +173,8 @@ session.connect(function(err, result) {
       this.wsMessageHandler(event.data);
     });
 
-    this.wsConnection.addEventListener('error', (err) => {
-      return dfd.reject(err);
+    this.wsConnection.addEventListener('error', () => {
```
Comment (vocoded) [inline sdks/js/src/classes/session.js:176]: no argument is available?

File: sdks/js/src/vendor_opus-recorder_README.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
+# Opus & Wave Recorder
```
Comment (vocoded) [inline sdks/js/src/vendor_opus-recorder_README.md]: where did this come from?

File: sdks/js/src/classes/session.js [hunk 7]
```diff
@@ -172,8 +173,8 @@ session.connect(function(err, result) {
       this.wsMessageHandler(event.data);
     });
 
-    this.wsConnection.addEventListener('error', (err) => {
```
Comment (ihorserba) [inline sdks/js/src/classes/session.js:175]: There is no useful information here in `err` parameter.

File: sdks/js/src/vendor_opus-recorder_README.md (hunk 6 above)
Comment (ihorserba) [reply sdks/js/src/vendor_opus-recorder_README.md]: Already removed, commited by mistake

File: sdks/js/src/classes/session.js (hunk 5 above)
Comment (ihorserba) [reply sdks/js/src/classes/session.js:176]: This event here is useless - it provides no information about the error

Comment (vocoded) [reply sdks/js/src/classes/session.js:176]: i think it's still better to pass it along, perhaps in the future a better version of the library will improve it

**Review by vocoded — APPROVED** [review]

**Pull request merged**